### ISSUE TITLE
Compilation fails due to Catch2 library

### ISSUE THREAD

**Issue by Schweini07** [Protinon/Binary-Reader#2]
Hi, first of thanks a lot for your amazing library, it makes working with binary files really trivial :slightly_smiling_face:.

Unfortunately I only was able to use the library by removing the tests section from the CMakeLists.txt
Else I am getting these errors:
`Binary-Reader/test/Catch2/src/catch2/../catch2/internal/catch_string_manip.hpp:47:14: error: ‘uint64_t’ in namespace ‘std’ does not name a type; did you mean ‘wint_t’?`
`Binary-Reader/test/Catch2/src/catch2/../catch2/internal/catch_string_manip.hpp:51:42: error: expected ‘)’ before ‘count’`

Maybe this is because of an update in the Catch2 library breaking compatibility?

**Comment by Protinon** [comment]
Glad to hear you're finding it useful! Honestly this was a commit-and-forget repo I used for 1 project, so I haven't felt a need to keep it updated.

That being said, I've gladly patched these issues. Catch2 was indeed outdated, so I've pulled in the latest version (you'll need to run `git submodule update`). I've also made test compilation optional and off by default. That'll make it easier to add as a sub-project in CMake.
